types/node/v18/test.d.ts

```diff
@@ -59,31 +59,33 @@ declare module "node:test" {
      * @param options Configuration options for the suite
      * @param fn The function under suite. Default: A no-op function.
      */
-    function describe(name?: string, options?: TestOptions, fn?: SuiteFn): void;
-    function describe(name?: string, fn?: SuiteFn): void;
-    function describe(options?: TestOptions, fn?: SuiteFn): void;
-    function describe(fn?: SuiteFn): void;
+    function describe(name?: string, options?: TestOptions, fn?: SuiteFn): Promise<void>;
```

Comment:
This caused our linter to start failing with `@typescript-eslint/no-floating-promises`, as nobody awaits the calls to describe/it. Furthermore, actually awaiting the `it` calls causes the second `it` to be delayed, which in-turn causes an error. For now, I've turned off that rule for test files, but ideally test files would be covered with that rule.

Example failure: https://github.com/wixplosives/sample-monorepo/actions/runs/6304680777/job/17116535127?pr=711

Reply:
See https://github.com/typescript-eslint/typescript-eslint/issues/7008